torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -125,7 +125,7 @@ def compute_q(fut):
         return [
             dist.all_reduce(q, group=group_to_use, async_op=True)
             .get_future()
-            .value()[0]
+            .wait()[0]
```

Comment:
This is needed because, in the PythonFutureWrapper, calling wait() also returns a value(), whereas calling value() does _not_ wait. This was already true on the CPU side (i.e., value() fails if the future isn't complete), and now it's also true for the GPU part (we don't sync streams when calling value()).

This was a discrepancy between ivalue::Future and FutureNCCL which I think we should fix.

The value() method is thus supposed to only be used to retrieve the value once we know it's already been correctly waited/synchronized, for example within a callback.